File: responses/YTT5DJ-9VR/3_find_maximum_product.py
# you can write to stdout for debugging purposes, e.g.
# print("this is a debug message")

def solution(A):
    if len(A)==0:
        return ""
    
    pref = A[0]
    for s in A:
        while pref != s[:len(pref)]:
            pref = pref[:-1]
            #if len(pref)==0:
            #    return ""
    return pref
    pass

File: responses/H3G23A-8H4/3_find_maximum_product.rb
# you can write to stdout for debugging purposes, e.g.
# puts "this is a debug message"

def solution(x, b)
  return 0 if b.length == 0 || x > b.length
  
  c = b[0]
  d = b[1]
  b.each_with_index do |i, index|
    (index..x).each do  |j|
      if (b[index] * b[j] > c * d)
        c = b[index]
        d = b[j]
      end
    end
  end
  p c 
  p d
  c * d
end